### PR TITLE
[No QA] Simplify and upgrade KeyboardShortcut library

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
This PR branches off of https://github.com/Expensify/App/pull/7702 to extract the group of changes that are related to the KeyboardShortcut library.

It also goes a bit further to simplify and clarify the KeyboardShortcut library API.

### Fixed Issues
$ (partial) https://github.com/Expensify/App/issues/7648

### Tests (web/desktop)
1. Press `CMD+I` to open the keyboard shortcut modal.
1. Press `Esc` to close it.
1. Press `CMD+K` to open the search page.
1. Press `Esc` to close it.
1. Press `CMD+SHIFT+K` to open the New Group page.
1. Press `Esc` to close it.
1. Press the avatar to open the settings page.
1. Press `Esc` to close it.
1. Highlight some text in a chat, then press `CMD+C`.
1. Paste the text somewhere and verify that it was correctly copied to the clipboard.

### Tests (native)
The KeyboardShortcut library is a noop on native. Regression tests only.

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I verified the PR has a small number of commits behind `main`
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I clearly indicated the environment tests should be run in (Staging vs Production)
- [x] I wrote testing steps that cover success & fail scenarios (if applicable)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests & verify they passed on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I added comments when the code was not self explanatory
    - [x] I put all copy / text shown in the product in all `src/languages/*` files (if applicable)
    - [x] I followed proper naming convention for platform-specific files (if applicable)
    - [x] I followed style guidelines (in [`Styling.md`](https://github.com/Expensify/App/blob/main/STYLING.md)) for all style edits I made
    - [x] I followed the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs))
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I corroborated the UI performance was not affected (the performance is the same than `main` branch)
- [x] If I created a new component I verified that a similar component doesn't exist in the codebase

#### PR Reviewer Checklist
- [ ] I verified the PR has a small number of commits behind `main`
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the testing environment is mentioned in the test steps
- [ ] I verified testing steps cover success & fail scenarios (if applicable)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors related to changes in this PR
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified comments were added when the code was not self explanatory
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [ ] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [ ] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components are not impacted by changes in this PR (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I verified the UI performance was not affected (the performance is the same than `main` branch)
- [ ] If a new component is created I verified that a similar component doesn't exist in the codebase

### QA Steps
Regression tests only.

- [ ] Verify that no errors appear in the JS console

#### Web
https://user-images.githubusercontent.com/47436092/157535621-0f1b9c06-3d8d-43ff-a29e-de096691fcc2.mov

#### Desktop
https://user-images.githubusercontent.com/47436092/157535712-d3e6181e-1531-457c-9493-65e5c282373c.mov

### CODE REVIEW

**Comment by roryabraham** [comment]
Tagging in @rushatgabhane for a review since you already reviewed most of these changes in https://github.com/Expensify/App/pull/7702

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
Thanks for tagging me
@roryabraham could you please pull the changes to Button.js from #7702?

I'm guessing it'll be needed by https://github.com/Expensify/App/pull/7920#discussion_r816054768 and https://github.com/Expensify/App/issues/7623

Otherwise, looks good. Liking the change to abstract away the modifiers

**Comment by roryabraham** [comment]
Updated to include changes in `Button.js`

File: src/components/Button.js [hunk 1]
```diff
@@ -48,6 +48,10 @@ const propTypes = {
     /** Call the onPress function when Enter key is pressed */
     pressOnEnter: PropTypes.bool,
 
+    /** The priority to assign the enter key event listener. 0 is the highest priority. */
+    enterKeyEventListenerPriority: PropTypes.number,
+
+
```
Comment (rushatgabhane) [inline src/components/Button.js:54]: Extra line breaker. (We really need a eslint rule for this)

**Review by rushatgabhane — APPROVED** [review]
@thienlnam LGTM! 🎉 and tests well. All yours!

**Comment by rushatgabhane** [comment]
Commenting since I can't edit the PR
#### PR Reviewer Checklist
- [x] I verified the PR has a small number of commits behind `main`
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the testing environment is mentioned in the test steps
- [x] I verified testing steps cover success & fail scenarios (if applicable)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified comments were added when the code was not self explanatory
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [x] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [x] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components are not impacted by changes in this PR (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified the UI performance was not affected (the performance is the same than `main` branch)
- [ ] If a new component is created I verified that a similar component doesn't exist in the codebase

File: src/components/Button.js (hunk 1 above)
Comment (roryabraham) [reply src/components/Button.js]: Is there somewhere where we've explicitly added this to the review guidelines?

Comment (roryabraham) [reply src/components/Button.js]: Because we can easily turn on https://eslint.org/docs/rules/no-multiple-empty-lines

Comment (rushatgabhane) [reply src/components/Button.js]: > Is there somewhere where we've explicitly added this to the review guidelines

Nope.

Yeah, let's add that rule.

Comment (rushatgabhane) [reply src/components/Button.js]: @roryabraham I'm gonna create a quick PR for [eslint-config-expensify](https://github.com/Expensify/eslint-config-expensify) to turn it on.

**Review by thienlnam — APPROVED** [review]
No complaints on the changes, it all looks good - just had one question

File: src/CONST.js [hunk 2]
```diff
@@ -176,8 +176,9 @@ const CONST = {
         },
     },
     KEYBOARD_SHORTCUT_KEY_DISPLAY_NAME: {
-        CONTROL: 'Ctrl',
-        META: 'Cmd',
+        CONTROL: 'CTRL',
+        ESCAPE: 'ESC',
+        META: 'CMD',
```
[the remark this answers is not in the document]
Comment (roryabraham) [reply src/CONST.js:181]: IDK

**Comment by marcaaron** [comment]
Seeing some console errors on dev I think might be related to these changes...

**Comment by marcaaron** [comment]
Oh sorry I was on **_staging_** not dev when this happened

**Comment by rushatgabhane** [comment]
Interesting. `Keydown` event is fired on clicking chrome's auto complete too.

https://user-images.githubusercontent.com/29673073/158280424-3123a6be-1c1f-4ddd-af11-839f651fd017.mov

**Comment by rushatgabhane** [comment]
One differentiating factor: Key presses are an `instanceof KeyboardEvent`, while the autocomplete thing is NOT.

**Pull request merged**